xbmc/epg/EpgContainer.cpp

```diff
@@ -233,20 +234,27 @@ void CEpgContainer::LoadFromDB(void)
   m_bLoaded = bLoaded;
 }
 
-bool CEpgContainer::PersistTables(void)
+bool CEpgContainer::PersistTables(bool bPostOnly /* = true */)
 {
-  m_critSection.lock();
-  auto copy = m_epgs;
-  m_critSection.unlock();
+  std::map<unsigned int, CEpg*> copy;
+  {
+    CSingleLock lock(m_critSection);
+    if (bPostOnly) //Persist tables only if db is disabled as persistAll do already this job here.
+      return m_bCallPersistTables = !CSettings::Get().GetBool("epg.ignoredbforclient");
+    else
+      copy = m_epgs;
+  }
   return m_database.Persist(copy);
```

Comment:
As preferred by @opdenkamp we might use a single exit point in methods (pvr section only).
We should also rename the parameter to `bMarkToPersist` to be a bit clearer what it does. And please add a method comment in the header.

```
bool CEpgContainer::PersistTables(bool bMarkToPersist /* = true */)
{
  bool bReturn(false);
 
  if (bMarkToPersist)
  {
    CSingleLock lock(m_critSection);
    // mark to persist only if db is enabled
    m_bCallPersistTables = !CSettings::Get().GetBool("epg.ignoredbforclient");
    bReturn = m_bCallPersistTables;
  }
  else
  {
    std::map<unsigned int, CEpg*> copy;
    {
      CSingleLock lock(m_critSection);
      copy = m_epgs;
    }
    bReturn = m_database.Persist(copy)
  }

 return bReturn;
}
```